cdap-ui/app/directives/dag-plus/my-dag-ctrl.js

```diff
@@ -439,6 +439,18 @@ angular.module(PKG.name + '.commons')
           from: conn.sourceId,
           to: conn.targetId
         });
+
+        if(!vm.isDisabled && !conn.getOverlay(['detachLabel' + conn.targetId])) {
+          conn.addOverlay(['Label', {
+              label: '<span class="fa fa-times connection-detach"></span>',
+              id: 'detachLabel' + conn.targetId,
+              location: 0.5,
```

Comment:
location 0.5 might not be ideal as users will definitely fall into this situation

<img width="1395" alt="screen shot 2017-03-24 at 5 54 18 pm" src="https://cloud.githubusercontent.com/assets/1452845/24317938/f6f38e00-10ba-11e7-8607-e4b40cf04bf2.png">

I am not sure what the middle `x` button will delete.